lib/urbanopt/reopt/reopt_ghp_adapter.rb

```diff
@@ -0,0 +1,326 @@
+          total_electric_load_building = []
+          # Ensure the column exists
+          if modelica_data.headers.include?(heating_power)
+            heating_power_values = modelica_data[heating_power]
+          end
+          if modelica_data.headers.include?(cooling_power)
+            cooling_power_values = modelica_data[cooling_power]
+          end
+          if modelica_data.headers.include?(pump_power)
+            pump_power_values = modelica_data[pump_power]
+          end
+          if modelica_data.headers.include?(ets_pump_power)
+            ets_pump_power_values = modelica_data[ets_pump_power]
+          end 
+
+          total_electric_load_building = heating_power_values.zip(cooling_power_values, pump_power_values, ets_pump_power_values).map do |elements|
+            elements.map { |e| e.to_f / 1000 }.sum
+          end
+
+          peak_combined_heatpump_thermal_ton = 0
+
+          if modelica_data.headers.include?(heating_system_capacity)
+            heating_system_capacity_value = modelica_data[heating_system_capacity][0]
+          end
+          if modelica_data.headers.include?(cooling_system_capacity)
+            cooling_system_capacity_value = modelica_data[cooling_system_capacity][0]
+          end
+
+          peak_combined_heatpump_thermal_ton = ([heating_system_capacity_value.to_f.abs, cooling_system_capacity_value.to_f.abs].max)/3517
+
+          # Store the result in reopt_inputs_building ElectricLoad
+          reopt_inputs_building[:ElectricLoad][:loads_kw] = total_electric_load_building
+
+
+
+          domestic_hot_water = total_electric_load_building.map do |load|
+            load * 0
+          end
+
+          # This is not used in REopt calculation but required for formatting.
+          reopt_inputs_building[:DomesticHotWaterLoad][:fuel_loads_mmbtu_per_hour] = domestic_hot_water
+
+          # Add GHP Fields
+          reopt_inputs_building[:GHP] = {}  
+          # REopt default
+          reopt_inputs_building[:GHP][:require_ghp_purchase] = 1
+          reopt_inputs_building[:GHP][:om_cost_per_sqft_year] = 0
+          reopt_inputs_building[:GHP][:heatpump_capacity_sizing_factor_on_peak_load] = 1.0
+          # Add the floor area
+          building_json_path = File.join(run_dir, building_id.to_s, "feature_reports", "default_feature_report.json")          
+          if File.exist?(building_json_path)
+            File.open(building_json_path, 'r') do |file|
+              building_json_data = JSON.parse(file.read, symbolize_names: true)
+              reopt_inputs_building[:GHP][:building_sqft] = building_json_data[:program][:floor_area_sqft]
+            end 
+          else
+            puts "File not found: #{building_json_path}"
+          end
+
+          # Add existing boiler fuel cost
+          # TODO : Add this as optional user input
+          reopt_inputs_building[:ExistingBoiler][:fuel_cost_per_mmbtu] = 13.5
```

Comment:
where did this fuel cost number come from?

Reply:
This is the default for the cost of Natural Gas per mmbtu, used in the REopt LCCA